nixos/modules/services/backup/syncoid.nix

```diff
@@ -170,47 +181,56 @@ in {
     # Implementation
 
     config = mkIf cfg.enable {
-      users =  {
+      users = {
         users = mkIf (cfg.user == "syncoid") {
           syncoid = {
             group = cfg.group;
             isSystemUser = true;
+            # For syncoid to be able to create /var/lib/syncoid/.ssh/
+            # and to use custom ssh_config or known_hosts.
+            home = "/var/lib/syncoid";
+            createHome = true;
```

Comment:
Sure, I've move to `StateDirectory=` which ensures correct permissions and automatically binds the directory in `RootDirectory=`.

Concerning the use case of `/var/lib/syncoid/.ssh/`, when generating a `/etc/ssh/ssh_known_hosts` with:
```
programs.ssh.knownHosts = {
  mermet = {
    hostNames = [ "mermet" "mermet.sourcephile.fr" ];
    publicKey = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIFvKN2sIpH782MFjaOpcty1Hs/T/TPNJpXI08H3O3oxl";
  };
};
```
and using this hostname in :
```
services.syncoid.commands."backup@mermet.sourcephile.fr:rpool/var/mail".target =
  "losurdo/backup/mermet/var/mail";
```
`syncoid` now writes a `/var/lib/syncoid/.ssh/known_hosts` specifying the IP address of the hostname:
```
80.67.180.129 ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIFvKN2sIpH782MFjaOpcty1Hs/T/TPNJpXI08H3O3oxl
```
instead of issuing the following warnings:
```
losurdo syncoid[19440]: Could not create directory '/var/empty/.ssh'.
losurdo syncoid[19440]: Failed to add the ED25519 host key for IP address '80.67.180.129' to the list of known hosts (/var/empty/.ssh/known_hosts).
```
Hence, if that IP address is dynamic, having such a stateful `known_hosts` may be useful, though there are probably some `--sshoption` that can help.